core/zoe/docs/zoe.md

```diff
@@ -0,0 +1,263 @@
+## An example: A swap
+
+I want to trade my three bricks for five wool. You realize you have
+five wool and agree to the deal. Without Zoe, though, you might send
+me the five wool, and I might disappear without ever giving you the
+three bricks in return. With Zoe, we can safely trade with each
+other, even if we don't trust one another. We can also safely rely on
+the swap smart contract. At worst, if the swap contract behaves badly,
+we will both get a refund, and at best, we'll get what we each wanted.
+
+Let's use the basic `publicSwap` contract ([full text of
+the real contract](/core/zoe/contracts/publicSwap.js)). 
+
+Here's a high-level overview of what would happen:
+1. I make an instance of the swap contract.
+2. I escrow my three bricks with Zoe and get an escrow receipt and a
+   promise for a payout in return.
+3. I send my escrow receipt to the swap as the first offer.
+4. I tell you the swap's `instanceId`
+5. Using the `instanceId`, you look up the swap with Zoe.
+6. You verify that it's using the `publicSwap` contract
+   code you expect, and can ask the swap about the offers made so far.
+7. You escrow your offer (offering five wool for three bricks) with
+   Zoe, getting an escrow receipt and a promise for a payout in
+   return.
+8. You send your escrow receipt to the swap as a matching offer.
+9. The offer matches and both of our payout promises resolve to [ERTP
+   payments](TODO), mine to the five wool that I wanted, and yours to
+   the three bricks that you wanted. Success!
+
+
+## How to write smart contracts
+
+Writing smart contracts that run on Zoe is easy, but let's look
+at a simple contract. This contract only does one thing, and
+it's pretty useless - it gives you back what you put in. Let's call it
+`automaticRefund`. Let's say the code of `automaticRefund` looks like this (see
+the [real contract code here](/core/zoe/contracts/automaticRefund.js)):
+
+```js
+export const makeContract = (zoe, terms) => {
+  return {
+    instance: {
+      makeOffer: async escrowReceipt => {
+        const { id: offerId} = await zoe.burnEscrowReceipt(escrowReceipt);
+        zoe.complete([offerId]);
+      },
+    },
+    assays: terms.assays,
+  };
+};
+```
+(In a real contract, whenever we create a new object or array, we recursively
+deep-freeze it with `@agoric/harden`. You can [learn more about `harden` here](https://github.com/Agoric/harden).)
+
+`automaticRefund` has one method exposed to the user: `makeOffer`.
+`makeOffer` takes in a `escrowReceipt`, and after burning the
+`escrowReceipt` (and thus verifying it as well), it tells Zoe to
+complete the offer and send the user their payout. 
+
+A smart contract on Zoe must export a function `makeContract` that
+takes two parameters: `zoe`, which is the governing contract facet of
+Zoe, and `terms`, which are the contract terms that a contract
+instance is made with. `Terms` must include a property called
+`assays`, which is an array of assays, the public faces of mints. For
```

Comment:
No suggested change here. But noting how weird this is. To me, the mint is the private/admin face[t] of the assay/issuer.

You're using both "facet" and "face". Pick one.